core/src/main/java/org/frankframework/mongodb/JndiMongoClientFactory.java

```diff
@@ -15,41 +15,64 @@
 */
 package org.frankframework.mongodb;
 
-import java.util.List;
-import java.util.Properties;
+import java.util.concurrent.TimeUnit;
 
+import org.apache.commons.lang3.StringUtils;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
 import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
 
+import org.frankframework.jdbc.datasource.FrankResource;
 import org.frankframework.jdbc.datasource.ObjectFactory;
 import org.frankframework.util.AppConstants;
+import org.frankframework.util.CredentialFactory;
 
 /**
- * MongoClientFactory that retrieves its configuration from JNDI.
+ * MongoClientFactory that retrieves its configuration from either JNDI or `resources.yml`.
  *
  * @author Gerrit van Brakel
  */
-public class JndiMongoClientFactory extends ObjectFactory<MongoClient, MongoClient> implements IMongoClientFactory {
+public class JndiMongoClientFactory extends ObjectFactory<MongoClient, Object> {
```

Comment:
Kan `Jndi` dan niet beter uit de naam weg?

Reply:
OK, maar voor backports is het al niet handig iets aan de LarvaActions classes te doen! 😬  Die bestaan nog niet in 9.0 en ouder!

Ik snap in ieder geval wat de redenering is.